ignite/cmd/ignite/main.go

```diff
@@ -4,22 +4,41 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"os"
-
 	ignitecmd "github.com/ignite/cli/ignite/cmd"
 	chainconfig "github.com/ignite/cli/ignite/config/chain"
 	"github.com/ignite/cli/ignite/pkg/clictx"
 	"github.com/ignite/cli/ignite/pkg/cliui/colors"
 	"github.com/ignite/cli/ignite/pkg/cliui/icons"
+	"github.com/ignite/cli/ignite/pkg/gacli"
 	"github.com/ignite/cli/ignite/pkg/validation"
 	"github.com/ignite/cli/ignite/pkg/xstrings"
+	"os"
+	"strings"
 )
 
 func main() {
 	os.Exit(run())
 }
 
 func run() int {
+	defer func() {
+		if r := recover(); r != nil {
+			addCmdMetric(metric{
+				err:     fmt.Errorf("%v", r),
+				command: strings.Join(os.Args, " "),
+			})
+			fmt.Println(r)
+			os.Exit(1)
+		}
+	}()
```

Comment:
What do you think about moving the constants to the top so `exitCodeError` is also used in the recover?

```suggestion
	const (
		exitCodeOK    = 0
		exitCodeError = 1
	)

	defer func() {
		if r := recover(); r != nil {
			addCmdMetric(metric{
				err:     fmt.Errorf("%v", r),
				command: strings.Join(os.Args, " "),
			})
			fmt.Println(r)
			os.Exit(exitCodeError)
		}
	}()
```

Also, there might be an issue with the deferred `addCmdMetric()` call where the error won't be added to the metrics because the CLI might exit before the `gaclient.Send()` finishes. Probably in this case we should wait for the gopher to finish before the `addCmdMetric` finishes, maybe something to have in mind for other commands that might finish before the metric is sent. Otherwise the tracked data will be partial.